src/framework/control.ml

```diff
@@ -278,7 +278,11 @@ struct
 
     GU.global_initialization := true;
     GU.earlyglobs := get_bool "exp.earlyglobs";
-    Spec.init ();
+    if get_string "load_run" <> "" then (
+      Spec.init (Some (Serialize.unmarshal (Filename.concat (get_string "load_run") "spec_marshal")))
+    )
+    else
+      Spec.init None;
```

Comment:
`Spec.init` is much earlier than the rest of unmarshaling, not sure if this matters.

Reply:
In principle it should not, just be aware that if this marshaled data contains e.g. references they will not point to the same things after marshaling in as references e.g. in the marshaled in hash table, even if they did when it was marshaled out (which has also related to the issue with hashconsing), but hopefully no analysis has such things in its hash table.

Reply:
Well, the marshaling for analysis-specific data is very much related to what's in the solution. So this is a very relevant point.

For mallocWrapper hashtable it doesn't matter that the physical objects are not the same, just that both unmarshals separately get objects which have matching identity (by `equal`/`hash`).

So following this logic, all marshaled data should be put in a single file using a wrapping record, such that their inter-references are preserved properly. And that no blocks need to be duplicated on disk or after unmarshaling.